src/app/services/generate-csv.service.ts

```diff
@@ -50,4 +55,27 @@ export class GenerateCsvService {
     }
     return str;
   }
+
+  ConvertAssetReadsToCSV(assetData: any) {
+    let str = '';
+    let row = 'timestamp';
+    const keys = Object.keys(assetData[0].reading);
+    for (const header of keys) {
+      row += ',' + header;
+    }
+    row = row.slice(0, -1);
```

Comment:
remove `slice(0, -1);` as row has default value  timestamp and `row += ',' + header;`